tests/i18n/tests.py

```diff
@@ -1859,6 +1859,11 @@ def test_page_with_dash(self):
         response = self.client.get('/de-simple-page/')
         self.assertEqual(response.content, b'Yes')
 
+    # Regression test for #27402
+    def test_404_page_no_default_language_redirect(self):
```

Comment:
What about `test_no_redirect_on_404`?
Can you try to shortly describe the problem in a docstring so that the next person reading the test does not have to open the ticket?